app/services/check_answers/relevant_section_finder_service.rb

```diff
@@ -0,0 +1,28 @@
+module CheckAnswers
+  class RelevantSectionFinderService
+    def self.call(step)
+      new.call(step)
+    end
+
+    def call(step)
+      data = SectionListerService.call
+      relevant_section = data[:sections].find do |section|
+        section[:screen] == step.to_s || matching_subsection?(section, step)
+      end
+
+      relevant_section&.dig(:label)
+    end
+
+    def matching_subsection?(section, step)
+      section[:subsections]&.any? do |subsection|
+        subsection[:screen] == step.to_s || matching_field?(subsection, step)
+      end
+    end
+
+    def matching_field?(subsection, step)
+      subsection[:fields]&.any? do |field|
+        field[:screen] == step.to_s
+      end
+    end
+  end
+end
```

Comment:
I'm feeling that this data (that used to be owned by one module/class) has now 'escaped' into another module/class, and the old  'service' does nothing apart from loading the YML file. 
Could it not be re-organised into methods rather than hash access? [Sorry can't think of a better suggestion at the moment]

Reply:
You're right, we now have a Presenter and a Service that both know about an idiosyncratic data structure. I'll have a ponder.